runtime/src/status_cache.rs

```diff
@@ -230,7 +230,21 @@ impl<T: Serialize + Clone> StatusCache<T> {
                 (
                     *slot,
                     self.roots.contains(slot),
-                    self.slot_deltas.get(slot).unwrap_or(&empty).clone(),
+                    Arc::clone(self.slot_deltas.get(slot).unwrap_or(&empty)),
+                )
+            })
+            .collect()
+    }
+
+    /// Get the statuses for all the root slots
+    pub fn root_slot_deltas(&self) -> Vec<SlotDelta<T>> {
+        self.roots
+            .iter()
+            .map(|slot| {
+                (
+                    *slot,
+                    true,
+                    self.slot_deltas.get(slot).cloned().unwrap_or_default(),
```

Comment:
why `unwrap_or_default()` is chosen instead of the `let empty = ..` in `fn slot_deltas(...)`?

I mean, this is slightly changing semantics for missing case. `fn slot_deltas()` points to the same instance (`empty`) for missing slots. but `fn root_slot_deltas` points to different instances. Assuming the caller is snapshot, it's not conceivable to modify the returned object. but there's no guarantee of future use with mutation. so, i'd like to see some clarification.

or, is it guaranteed that if slot is in self.roots, `self.slot_deltas` should have that slot too?. in that case, I'd like `.expect()` or just plain `unwrap()`.

Reply:
> Other than slot 0, can valid blocks be rooted that have zero transactions?

@brooksprumo fyi, i learned that this is surely possible: https://explorer.solana.com/block/146016008 (this has no transactions)